pandas/core/reshape/melt.py

```diff
@@ -121,7 +121,19 @@ def melt(
         # asanyarray will keep the columns as an Index
         mdata[col] = np.asanyarray(frame.columns._get_level_values(i)).repeat(N)
 
-    return frame._constructor(mdata, columns=mcolumns)
+    result = frame._constructor(mdata, columns=mcolumns)
+
+    if not ignore_index:
+        new_index = np.tile(frame.index, K)
```

Comment:
both MI and Index already have a .repeat() method, I think we could add a .tile() method to make this easier. (or just use repeat)

Reply:
`Index(["foo", "bar"]).repeat(2)` yields `Index(['foo', 'foo', 'bar', 'bar'], dtype='object')`, where as `np.tile(["foo", "bar"])` yields `array(['foo', 'bar', 'foo', 'bar', 'foo', 'bar'], dtype=object)`. The latter corresponds to the layout used in `melt` so it's very not trivial to use repeat instead of tile.

I tried having a look at implementing `tile` on indices but then I would also have to do it for multiindices and document it and tests, and argument validation which I've never even looked at before and I think it's a big hassle that I will not undertake.

Reply:
ok fair enough

Reply:
we already have _tile_compat in pandas\core\reshape\util.py. This may allow futher simplification here.

Reply:
This converts to object dtype. Can you use `pandas.core.reshape.util._tile_compat`?:

Reply:
You should be able to remove the next section as well, since you won't be converting to an ndarray.

Reply:
Thanks, that simplifies the code a lot.